features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/SystemNotificationsEnabledProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.preferences.impl.notifications
+
+import android.content.Context
+import androidx.core.app.NotificationManagerCompat
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.libraries.di.AppScope
+import io.element.android.libraries.di.ApplicationContext
+import io.element.android.libraries.di.SingleIn
+import javax.inject.Inject
+
+interface SystemNotificationsEnabledProvider {
```

Comment:
Is this actually needed? It can be useful if we plan to use it elsewhere in the future, but I think you could have gotten the same result by using:

```kotlin
val context = LocalContext.current
val areNotificationsEnabled by remember {
    NotificationManagerCompat.from(context).areNotificationsEnabled()
}
```

Reply:
I see it's called again later. We could still remember the `NotificationManagerCompat` object and call that instead, but feel free to ignore this comment if keeping the provider is easier.

Reply:
Are you suggesting to put it in the view rather than the presenter?

Felt more like something that should be in the presenter and just pass data to the view, but then to keep the presenter testable I didn't want it to have a reference to the context.